Sources/Core/Extensions/Foundation/URL+PovioKit.swift

```diff
@@ -32,9 +34,24 @@ public extension URL {
     components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
     return components.url ?? absoluteURL
   }
+  
+  /// Retrieves the query parameters from the URL as a dictionary.
+  ///
+  /// - Returns: An optional dictionary where the keys are `AnyHashable` representing the query parameter names,
+  ///            and the values are `Any` representing the corresponding query parameter values.
+  ///            Returns nil if the URL is invalid or has no query parameters.
+  var queryParameters: [AnyHashable: Any]? {
+    guard let components = URLComponents(url: self, resolvingAgainstBaseURL: true),
+          let queryItems = components.queryItems else { return nil }
+    var params: [AnyHashable: Any] = [:]
+    queryItems.forEach { queryItem in
+      queryItem.value.map { params[queryItem.name] = $0 }
+    }
+    return params
+  }
 }
 
-extension URL: ExpressibleByStringLiteral {
+extension URL: @retroactive ExpressibleByStringLiteral {
```

Comment:
Can we use `Swift.ExpressibleByStringLiteral` so this code can still build on older swift versions

Reply:
Tbh, I don't like it. And regarding to the [documentation](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0364-retroactive-conformance-warning.md#source-compatibility), we can do it differently.